cterasdk/edge/firmware.py

```diff
@@ -0,0 +1,55 @@
+from ..lib import FileSystem
+from ..exception import CTERAException
+from .base_command import BaseCommand
+
+
+class UploadTaskStatus():
+    IN_PROGRESS = 0
+    COMPLETE = 1
+    FAIL = -1
+
+
+class Firmware(BaseCommand):
+    """ Gateway Firmware upgrade API """
+
+    def __init__(self, gateway):
+        super().__init__(gateway)
+        self._filesystem = FileSystem.instance()
+
+    def upgrade(self, file_path, reboot=True, wait_for_reboot=True):
```

Comment:
When uploading a firmware directly to the edge filer, it automatically reboots. I'm not sure the ``reboot=True`` flag is necessary.

Reply:
The UI calls the reboot API. If you don't call it, then the new F/W just sits there waiting for the next reboot